libs/stats/odc/stats/utils.py

```diff
@@ -170,3 +177,87 @@ def dedup_s2_datasets(dss):
         out.append(chunk[-1])
         skipped.extend(chunk[:-1])
     return out, skipped
+
+
+def fuse_products(type_1: DatasetType, type_2: DatasetType, dc: Datacube):
+    """
+    Fuses two products. This function requires access to a Datacube to access the metadata type.
+    
+    Fusing two products requires that:
+      - both metadata types are eo3
+      - there are no conflicting band names
+      - the file formats are identical
+    """
+
+    def_1, def_2 = type_1.definition, type_2.definition
+    fused_def = dict()
+
+    assert def_1["metadata_type"] == def_2["metadata_type"]
+    assert def_1["metadata_type"] == "eo3"
+
+    measurements_1 = set(m["name"] for m in def_1['measurements'])
+    measurements_2 = set(m["name"] for m in def_2['measurements'])
+    assert len(measurements_1.intersection(measurements_2)) == 0
+
+    file_format = def_1["metadata"]["properties"]["odc:file_format"]
+    assert file_format == def_2["metadata"]["properties"]["odc:file_format"]
+
+    name = f"fused__{def_1['name']}__{def_2['name']}"
+
+    fused_def["name"] = name
+    fused_def["metadata"] = {"product": {"name": name}, "properties": {"odc:file_format": file_format}}
+    fused_def["description"] = f"Fused products: {def_1['description']}, {def_2['description']}"
+    fused_def["measurements"] = def_1["measurements"] + def_2["measurements"] 
+    fused_def["metadata_type"] = def_1["metadata_type"]
+
+    return dc.index.products.from_doc(fused_def)
```

Comment:
This requires connection to db, not ideal. 
It only needs this to lookup metadata_type document from db using `metadata_type` field from product spec, but we already have metadata_type, cause we have product objects. Can you please change this to construct `DatasetType` directly with whatever MetadataType object is supplied in the first product.